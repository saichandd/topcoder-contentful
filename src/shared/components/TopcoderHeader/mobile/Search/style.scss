@import '~styles/mixins';

.icon {
  left: 18px;
  position: absolute;
  top: 14px;
}

.search {
  padding: 6px 12px 0;
  position: relative;

  input {
    background: none;
    border: 1px solid $tc-gray-60;
    box-shadow: none;
    color: $tc-white;
    font: 16px "Sofia Pro", sans-serif;
    height: 32px;
    outline: none;
    padding-left: 30px !important;

    &::placeholder {
      color: $tc-gray-60;
    }

    &:hover {
      border: 1px solid $tc-gray-40;
    }

    &:focus {
      border: 1px solid $tc-dark-blue;
    }
  }

  input:active,
  input:focus,
  input:hover {
    box-shadow: none;
    outline: none;
  }
}
