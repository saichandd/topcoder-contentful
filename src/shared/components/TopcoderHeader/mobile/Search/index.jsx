import React from 'react';
import { config } from 'topcoder-react-utils';

import IconMagnifyingGlass from '../../../../../assets/images/magnifying_glass.svg';
import './style.scss';

/* NOTE: We use window object only inside the keypress handler, which only can
 * be invoked at the client side, thus the code is still isomorphic. */
/* global window */

export default function Search() {
  return (
    <div styleName="search">
      <IconMagnifyingGlass styleName="icon" />
      <input
        onKeyPress={(event) => {
          if (event.key === 'Enter') {
            window.location = `${config.URL.BASE}/search/members?q=${
              encodeURIComponent(event.target.value)
            }`;
          }
        }}
        placeholder="Find members by username or skill"
        aria-label="Find members by username or skill"
      />
    </div>
  );
}
