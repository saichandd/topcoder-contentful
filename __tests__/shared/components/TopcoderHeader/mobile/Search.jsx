import _ from 'lodash';
import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import Search from 'components/TopcoderHeader/mobile/Search';
import TU from 'react-dom/test-utils';

class SearchWrapper extends React.Component {
  componentDidMount() {}

  render() {
    return (
      <Search {...this.props} />
    );
  }
}

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  renderer.render(<Search />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

const page = TU.renderIntoDocument(<SearchWrapper />);

test.skip('Handles Enter key', () => {
  const items = TU.scryRenderedDOMComponentsWithClass(page, 'search');
  const input = _.find(items[0].children, item => item && item.tagName === 'INPUT');
  TU.Simulate.keyPress(input, {
    key: 'Enter',
    target: {
      value: 'SEARCH',
    },
  });
  /* TODO: We want to check here that /search/members?q=SEARCH
   * was put into window.location, but inside jest test window.location
   * comes from jsdom, and it is not really changed (at least,
   * checking window.location returns a weird data structure,
   * what is the way to access set url is not that clear).
   * Should be investigated. */
});
