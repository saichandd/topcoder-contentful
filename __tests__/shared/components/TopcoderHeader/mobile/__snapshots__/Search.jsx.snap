// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-TopcoderHeader-mobile-Search-___style__search___101Ux"
>
  <IconMagnifyingGlass
    className="src-shared-components-TopcoderHeader-mobile-Search-___style__icon___aCkZk"
    height="18"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  />
  <input
    aria-label="Find members by username or skill"
    onKeyPress={[Function]}
    placeholder="Find members by username or skill"
  />
</div>
`;
